src/WebJobs.Script/Host/BlobLeaseManager.cs

```diff
@@ -123,62 +126,48 @@ private void ProcessLeaseTimerTick(object state)
 
         private async Task AcquireOrRenewLeaseAsync()
         {
-            try
+            string lockName = GetBlobName(_hostId);
+
+            DateTime requestStart = DateTime.UtcNow;
+            if (HasLease)
             {
-                DateTime requestStart = DateTime.UtcNow;
-                if (HasLease)
+                try
                 {
-                    await _lockBlob.RenewLeaseAsync(new AccessCondition { LeaseId = LeaseId });
+                    await _lockManager.RenewAsync(this.LockHandle, CancellationToken.None);
+
                     _lastRenewal = DateTime.UtcNow;
                     _lastRenewalLatency = _lastRenewal - requestStart;
                 }
-                else
+                catch
                 {
-                    LeaseId = await _lockBlob.AcquireLeaseAsync(_leaseTimeout, _instanceId);
-                    _lastRenewal = DateTime.UtcNow;
-                    _lastRenewalLatency = _lastRenewal - requestStart;
-
-                    string message = $"Host lock lease acquired by instance ID '{_instanceId}'.";
-                    _traceWriter.Info(message);
-                    _logger?.LogInformation(message);
-
-                    // We've successfully acquired the lease, change the timer to use our renewal interval
-                    SetTimerInterval(_renewalInterval);
+                    // The lease was 'stolen'. Log details for debugging.
+                    string lastRenewalFormatted = _lastRenewal.ToString("yyyy-MM-ddTHH:mm:ss.FFFZ", CultureInfo.InvariantCulture);
+                    int millisecondsSinceLastSuccess = (int)(DateTime.UtcNow - _lastRenewal).TotalMilliseconds;
+                    int lastRenewalMilliseconds = (int)_lastRenewalLatency.TotalMilliseconds;
+                    ProcessLeaseError($"Another host has acquired the lease. The last successful renewal completed at {lastRenewalFormatted} ({millisecondsSinceLastSuccess} milliseconds ago) with a duration of {lastRenewalMilliseconds} milliseconds.");
                 }
             }
-            catch (StorageException exc)
+            else
             {
-                if (exc.RequestInformation.HttpStatusCode == 409)
+                // LockManager will handle various cases like 500s, 404s, retry if container doesn't exist, etc.
+                string proposedLeaseId = _instanceId;
+                this.LockHandle = await _lockManager.TryLockAsync(null, lockName, _instanceId, proposedLeaseId, _leaseTimeout, CancellationToken.None);
```

Comment:
For diagnostics, we stamp a lock with an "owner".  SDK uses this aggressively so that you know which function instance holds the lock.  Could be null in this case. 

---
In reply to: [123670650](https://github.com/Azure/azure-webjobs-sdk-script/pull/1615#discussion_r123670650) [](ancestors = 123670650)